app/demo-interactor/HostKNDemoRunner.cc

```diff
@@ -0,0 +1,213 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file HostKNDemoRunner.cc
+//---------------------------------------------------------------------------//
+#include "HostKNDemoRunner.hh"
+
+#include <iostream>
+#include <random>
+#include "base/Stopwatch.hh"
+#include "base/Range.hh"
+#include "base/ArrayUtils.hh"
+#include "random/distributions/ExponentialDistribution.hh"
+#include "physics/base/ParticleTrackView.hh"
+#include "physics/base/Units.hh"
+#include "physics/base/Secondary.hh"
+#include "physics/base/SecondaryAllocatorView.hh"
+#include "physics/em/KleinNishinaInteractor.hh"
+#include "PhysicsArrayCalculator.hh"
+#include "DetectorView.hh"
+#include "HostStackAllocatorStore.hh"
+#include "HostDetectorStore.hh"
+
+namespace demo_interactor_cpu
+{
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Construct with parameters.
+ */
+HostKNDemoRunner::HostKNDemoRunner(constSPParticleParams     particles,
+                                   constSPPhysicsArrayParams xs)
+    : pparams_(std::move(particles)), xsparams_(std::move(xs))
+{
+    REQUIRE(pparams_);
+    REQUIRE(xsparams_);
+
+    // Set up KN interactor data;
+    namespace pdg            = celeritas::pdg;
+    kn_pointers_.electron_id = pparams_->find(pdg::electron());
+    kn_pointers_.gamma_id    = pparams_->find(pdg::gamma());
+    kn_pointers_.inv_electron_mass
+        = 1 / pparams_->get(kn_pointers_.electron_id).mass.value();
+    ENSURE(kn_pointers_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * \brief Run given number of particles each for max steps.
+ */
+
+auto HostKNDemoRunner::operator()(demo_interactor::KNDemoRunArgs args)
+    -> result_type
+{
+    REQUIRE(args.energy > 0);
+    REQUIRE(args.num_tracks > 0);
+
+    // Initialize results
+    result_type result;
+    result.time.reserve(args.max_steps);
+    result.alive.reserve(args.max_steps + 1);
+    result.edep.reserve(args.max_steps);
+
+    // Start timer for overall execution and transport-only time
+    celeritas::Stopwatch total_time;
```

Comment:
And since we're in the `celeritas` repo you're always welcome to add a `using namespace celeritas` 😄

Reply:
Yeh, I generally prefer not putting `using` at file scope, even in `.cc` files.  I was planning to add them to the function scope, but decided it wasn't worth it.  I'll do that now though cause I think it adds to line-level readability.